lib/vra/client.rb

```diff
@@ -93,19 +93,17 @@ def authorized?
       return false if @bearer_token.value.nil?
 
       response = http_head("/identity/api/tokens/#{@bearer_token.value}", :skip_auth)
-      if response.code == 204
-        true
-      else
-        false
-      end
+      response.success?
     end
 
     def generate_bearer_token
       @bearer_token.value = nil
       validate_client_options!
 
-      response = http_post('/identity/api/tokens', FFI_Yajl::Encoder.encode(bearer_token_request_body), :skip_auth)
-      if response.code != 200
+      response = http_post('/identity/api/tokens',
+                           FFI_Yajl::Encoder.encode(bearer_token_request_body),
+                           :skip_auth)
+      unless response.success?
```

Comment:
Same comment as above - only a 200 is expected here, anything else should warrant some investigation.